compiler/AST/LoopExpr.cpp

```diff
@@ -732,8 +735,19 @@ static CallExpr* buildLoopExprFunctions(LoopExpr* loopExpr) {
   bool insideArgSymbol = isArgSymbol(loopExpr->parentSymbol) ||
                          isTypeSymbol(loopExpr->parentSymbol);
 
+  // The loop expression may receive additional vars via attributes applied
+  // to its variable. This is represented by enclosing the LoopExpr inside
+  // a "gpu attribute block". See if we need to handle that, as well.
+  BlockStmt* attrBlock = nullptr;
+  BlockStmt* primsFromAttrs = nullptr;
+  if ((attrBlock = findEnclosingGpuAttributeBlock(loopExpr))) {
+    // The primitives might be applied to several expressions at the same
+    // time, so we can't steal them, and need to copy.
```

Comment:
do you have a case where you see this behavior?

Reply:
Yes, easily.

```Chapel
var A = foreach (i, j) in zip(foreach ..., foreach ...)
```

In the future, something similar might be the case for promoted expressions.